test/integration/api/PayableExchangeIssueAPI.spec.ts

```diff
@@ -0,0 +1,316 @@
+      baseSetNaturalUnit = new BigNumber(10 ** 9);
+      baseSetToken = await deploySetTokenAsync(
+        web3,
+        core,
+        setTokenFactory.address,
+        componentAddresses,
+        componentUnits,
+        baseSetNaturalUnit,
+      );
+
+      // Create the Rebalancing Set
+      rebalancingUnitShares = DEFAULT_UNIT_SHARES;
+      rebalancingSetToken = await createDefaultRebalancingSetTokenAsync(
+        web3,
+        core,
+        rebalancingSetTokenFactory.address,
+        DEFAULT_ACCOUNT,
+        baseSetToken.address,
+        ONE_DAY_IN_SECONDS,
+      );
+
+      subjectEther = new BigNumber(10 ** 10);
+
+      // Generate exchange issue data
+      exchangeIssueSetAddress = baseSetToken.address;
+      exchangeIssueQuantity = new BigNumber(10 ** 10);
+      exchangeIssuePaymentToken = wrappedEtherMock.address;
+      exchangeIssuePaymentTokenAmount = subjectEther;
+      exchangeIssueRequiredComponents = componentAddresses;
+      exchangeIssueRequiredComponentAmounts = componentUnits.map(
+        unit => unit.mul(exchangeIssueQuantity).div(baseSetNaturalUnit)
+      );
+
+      subjectExchangeIssueData = {
+        setAddress: exchangeIssueSetAddress,
+        paymentToken: exchangeIssuePaymentToken,
+        paymentTokenAmount: exchangeIssuePaymentTokenAmount,
+        quantity: exchangeIssueQuantity,
+        requiredComponents: exchangeIssueRequiredComponents,
+        requiredComponentAmounts: exchangeIssueRequiredComponentAmounts,
+      };
+
+      await approveForTransferAsync(
+        [baseSetComponent],
+        SetTestUtils.ZERO_EX_ERC20_PROXY_ADDRESS,
+        zeroExOrderMaker
+      );
+
+      // Create 0x order for the component, using weth(4) paymentToken as default
+      zeroExOrder = await setUtils.generateZeroExSignedFillOrder(
+        NULL_ADDRESS,                                     // senderAddress
+        zeroExOrderMaker,                                 // makerAddress
+        NULL_ADDRESS,                                     // takerAddress
+        ZERO,                                             // makerFee
+        ZERO,                                             // takerFee
+        subjectExchangeIssueData.requiredComponentAmounts[0],         // makerAssetAmount
+        exchangeIssuePaymentTokenAmount,                  // takerAssetAmount
+        exchangeIssueRequiredComponents[0],               // makerAssetAddress
+        exchangeIssuePaymentToken,                        // takerAssetAddress
+        SetUtils.generateSalt(),                          // salt
+        SetTestUtils.ZERO_EX_EXCHANGE_ADDRESS,            // exchangeAddress
+        NULL_ADDRESS,                                     // feeRecipientAddress
+        SetTestUtils.generateTimestamp(10000),            // expirationTimeSeconds
+        exchangeIssuePaymentTokenAmount,                  // amount of zeroExOrder to fill
+      );
+
+      subjectExchangeOrder = [zeroExOrder];
```

Comment:
None of these use Kyber, it'd be nice to have one set up where the component token is pulled off the reserve.

Reply:
Yeah true. How would we set that up?